src/payform.coffee

```diff
@@ -436,37 +441,155 @@
 
+  eventList = {
+    cvcInput: [
+      {
+        eventName: 'keypress',
+        eventHandler: _eventNormalize(restrictNumeric),
+      },
+      {
+        eventName: 'keypress',
+        eventHandler: _eventNormalize(restrictCVC),
+      },
+      {
+        eventName: 'paste',
+        eventHandler: _eventNormalize(reFormatCVC),
+      },
+      {
+        eventName: 'change',
+        eventHandler: _eventNormalize(reFormatCVC),
+      },
+      {
+        eventName: 'input',
+        eventHandler: _eventNormalize(reFormatCVC),
+      },
+    ],
+
+    expiryInput: [
+      {
+        eventName: 'keypress',
+        eventHandler: _eventNormalize(restrictNumeric),
+      },
+      {
+        eventName: 'keypress',
+        eventHandler: _eventNormalize(restrictExpiry),
+      },
+      {
+        eventName: 'keypress',
+        eventHandler: _eventNormalize(formatCardExpiry),
+      },
+      {
+        eventName: 'keypress',
+        eventHandler: _eventNormalize(formatForwardSlashAndSpace),
+      },
+      {
+        eventName: 'keypress',
+        eventHandler: _eventNormalize(formatForwardExpiry),
+      },
+      {
+        eventName: 'keydown',
+        eventHandler: _eventNormalize(formatBackExpiry),
+      },
+      {
+        eventName: 'change',
+        eventHandler: _eventNormalize(reFormatExpiry),
+      },
+      {
+        eventName: 'input',
+        eventHandler: _eventNormalize(reFormatExpiry),
+      },
+    ],
+
+    cardNumberInput: [
+      {
+        eventName: 'keypress',
+        eventHandler: _eventNormalize(restrictNumeric),
+      },
+      {
+        eventName: 'keypress',
+        eventHandler: _eventNormalize(restrictCardNumber),
+      },
+      {
+        eventName: 'keypress',
+        eventHandler: _eventNormalize(formatCardNumber),
+      },
+      {
+        eventName: 'keydown',
+        eventHandler: _eventNormalize(formatBackCardNumber),
+      },
+      {
+        eventName: 'paste',
+        eventHandler: _eventNormalize(reFormatCardNumber),
+      },
+      {
+        eventName: 'change',
+        eventHandler: _eventNormalize(reFormatCardNumber),
+      },
+      {
+        eventName: 'input',
+        eventHandler: _eventNormalize(reFormatCardNumber),
+      },
+    ],
+
+    numericInput: [
+      {
+        eventName: 'keypress',
+        eventHandler: _eventNormalize(restrictNumeric),
+      },
+      {
+        eventName: 'paste',
+        eventHandler: _eventNormalize(restrictNumeric),
+      },
+      {
+        eventName: 'change',
+        eventHandler: _eventNormalize(restrictNumeric),
+      },
+      {
+        eventName: 'input',
+        eventHandler: _eventNormalize(restrictNumeric),
+      },
+    ],
+  }
+
   payform.cvcInput = (input) ->
-    _on(input, 'keypress', restrictNumeric)
-    _on(input, 'keypress', restrictCVC)
-    _on(input, 'paste',    reFormatCVC)
-    _on(input, 'change',   reFormatCVC)
-    _on(input, 'input',    reFormatCVC)
+    for evt in eventList.cvcInput
```

Comment:
What do you think about refactoring the loop for attaching the events via `_on`?

Reply:
@ddayguerrero sure thing! Can you clarify a little further on how you'd like me to refactor?

Reply:
Sure! 😊 I considered this request because I noticed a pattern in all the for loops defined.

I suggest predefining all inputs (e.g. expiryInput, cardNumerInput, etc.) inside an array or JSON object. Then we can iterate through each type and define the function which calls the loop and attaches the event listeners.

Let me know if this makes sense!

Reply:
@ddayguerrero hey I just did a variation of what you suggested. Let me know what you think now

Reply:
Thanks for the good work; this looks promising! 👍 I'm wondering if it worth renaming the API as the event attachment is implied when using the `{field}Input(input)` function.

```
 payform.cvcInput(input);
 payform.detachCvcInput(input);
```

to 

```
 payform.attachCvcInput(input);
 payform.detachCvcInput(input);
```

May be far fetched as it will force us to create a more significant release version. I'll wait for @jondavidjohn input on this solution as a whole.

Reply:
Yeah, I was a little weary on releasing a breaking change to the API given how many people currently use this project.

My initial idea was for it to be a minor release where we introduce new methods API but no breaking changes. But I'll definitely also defer to @jondavidjohn here and let him make the final decision around that.

It would be a quick change on my end, so if that's what we decide to do, I can amend the PR very swiftly.

Reply:
Yeah, let's hold off any any breaking changes for now, this should definitely be a minor bump, but I don't think a major bump is worth it simply for semantics.

Good thought on making them consistent, keep that in mind if we ever have to force a breaking change to circle back. 👍